fastlane/lanes/release.rb

```diff
@@ -0,0 +1,236 @@
+    )
+    # It would be good to update the action so that it can:
+    #
+    # - Use a custom commit message, so that we can differentiate between
+    #   WordPress and Jetpack
+    # - Have some sort of interactive mode, where the file is extracted and
+    #   shown to the user and they can either confirm and let the lane commit,
+    #   or modify it manually first and then run through the
+    #   show-confirm-commit cycle again
+    #
+    # In the meantime, we can make due with a duplicated commit message and the
+    # `print_release_notes_reminder` at the end of the lane to remember to make
+    # updates to the files.
+    extract_release_notes_for_version(
+      version: new_version,
+      release_notes_file_path: release_notes_source_path,
+      extracted_notes_file_path: extracted_release_notes_file_path(app: :jetpack)
+    )
+    ios_update_release_notes(new_version: new_version)
+
+    setbranchprotection(repository: GHHELPER_REPO, branch: "release/#{new_version}")
+    setfrozentag(repository: GHHELPER_REPO, milestone: new_version)
+    ios_check_beta_deps(podfile: File.join(PROJECT_ROOT_FOLDER, 'Podfile'))
+
+    print_release_notes_reminder
+  end
+
+  # Completes the final steps for the code freeze
+  #
+  #  - Generate `.strings` files from code then merge the other, manually-maintained `.strings` files with it
+  #  - Trigger the build of the first beta on CI
+  #
+  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
+  #
+  desc 'Completes the final steps for the code freeze'
+  lane :complete_code_freeze do |options|
+    ios_completecodefreeze_prechecks(options)
+    generate_strings_file_for_glotpress
+
+    UI.confirm('Ready to push changes to remote and trigger the beta build?') unless ENV['RELEASE_TOOLKIT_SKIP_PUSH_CONFIRM']
+    push_to_git_remote(tags: false)
+    trigger_beta_build(branch_to_build: "release/#{ios_get_app_version}")
+  end
+
+  # Creates a new beta by bumping the app version appropriately then triggering a beta build on CI
+  #
+  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
+  # @option [String] base_version (default: _current app version_) If set, bases the beta on the specified version
+  #                  and `release/<base_version>` branch instead of the current one. Useful for triggering betas on hotfixes for example.
+  #
+  desc 'Trigger a new beta build on CI'
+  lane :new_beta_release do |options|
+    ios_betabuild_prechecks(options)
+    download_localized_strings_and_metadata(options)
+    # FIXME: (2021.06.17) This is disabled because we currently have a >256 chars string which GlotPress truncates when exporting  the `.strings` files,
+    #   leading to incorrect key for it and (rightful) linter failure. We need to split that key into 2 smaller copies before we can re-enable this.
+    # ios_lint_localizations(input_dir: 'WordPress/Resources', allow_retry: true)
```

Comment:
👆 _Note to self: remember to re-enable this very soon, after updating the problematic `NSLocalizedString` call in the code — now that we support arbitrary keys which don't have to be the English-copy (via `NSLocalizedString(key, value: "english copy", …)`) and can thus avoid this issue._